proposals/large-crypto-blobs.md

```diff
@@ -0,0 +1,127 @@
+
+This has been a pain for some users, including team spark did not rely on libp2p for NFT project because it was too heavy. This can be a threat for the wide usage of libp2p in the web3 ecosystem.
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Web3 users want to create libp2p backed applications with only the code they use bundled
+- Web3 users want to provide smooth and fast experiences to their users
+- Web3 users do not want to have their apps taking X seconds to load
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+- Users can configure via libp2p configuration what cryptographic modules they are interested in using
+- Libp2p provides custom bundles for the most common use cases that users can rely on (out of the box usage)
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+- Libp2p can be configured according to the cryptographic needs
+- Libp2p provides ready to go bundles for typical use cases
+- A minimal libp2p node can be configured with a maximum of 150Kb (?).
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+
+#### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+#### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need?_
+
```

Comment:
- WASM backend was brought up other day, which I think is worth considering in alternatives

Reply:
- Work with partners (e.g. Igalia) to ship needed crypto primitives natively in browsers.

Reply:
@Gozala can you give some thoughts here on the timeframe for a usable implementation via this route? This seems like a really good investment to make but we need to be clear about expectations of pay-off time I think.
There's likely cumulative lag involved in the steps that might look something like: (a) identify narrow requirements and negotiate grant, (b) run through spec process to get agreement from browser vendors, (c) implement for (or partner with) browser vendors to get them native .. then (d) wait for releases and critical-mass adoption.

Reply:
> @Gozala can you give some thoughts here on the timeframe for a usable implementation via this route?

I thing it is impossible to make guesses without doing steps below, I'd expect it to be on years scale.

1. Having concrete list of crypto primitives.
1. Having preliminary discussions with browser vendors / standard bodies to asses reception to the idea.
1. Some analyses on the amount of effort it would require across different engines.

Cc @autonome and @lidel who worked with Igalia on protocol registration effort and have some insight on how long it took from idea to having things shipped. 

I also don't think we should not pursue other routes in favor of this, I think it would be wise to do both.

Reply:
Added these alternatives. Thanks for raising this @Gozala